From 164d0706563c4c037299c324d58a777be3ec240f Mon Sep 17 00:00:00 2001
From: Gou Hao <gouhao@uniontech.com>
Date: Wed, 9 Mar 2022 21:48:58 +0800
Subject: [PATCH] eulerfs: remove redundant calculations
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 164d0706563c4c037299c324d58a777be3ec240f
Modified-by-SEL: No


uniontech inclusion
category: cleanup
bugzilla: https://gitee.com/openeuler/kernel/issues/I4X47D?from=project-issue
CVE: NA

--------------------------------

The 'left' always is 0.
If it is not 0, it will 'goto out;' from the previous if judgment.

Reviewed-by: Yu Kuai <yukuai3@huawei.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 fs/eulerfs/dax.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/fs/eulerfs/dax.c b/fs/eulerfs/dax.c
index 9ec8ad713fd9..131d08f1c6c1 100644
--- a/fs/eulerfs/dax.c
+++ b/fs/eulerfs/dax.c
@@ -1172,8 +1172,8 @@ static ssize_t do_mapping_read(struct address_space *mapping,
 			goto out;
 		}
 
-		copied += (nr - left);
-		offset += (nr - left);
+		copied += nr;
+		offset += nr;
 		index += offset >> PAGE_SHIFT;
 		offset &= ~PAGE_MASK;
 	} while (copied < len);
-- 
2.33.0

